R/fiveyeargwsum-render.R

```diff
@@ -1,100 +1,151 @@
-fiveyeargwsumPlot <- function(data) {
+createfiveyeargwsumPlot <- function(reportObject){
+  #Rendering Options
   options(scipen=8)
-  plot_object <- createfiveyeargwsumPlot(data)
-  return(plot_object)
-}
-
-createfiveyeargwsumPlot <- function(data){
   
-  isInverted <- data$reportMetadata$isInverted
-  fiveyrData <- parseFiveYrData(data)
-  
-  #semantics for min/max are swapped on inverted plots
-  maxLabel <- "Max. Instantaneous"
-  minLabel <- "Min. Instantaneous"
-  if(isInverted) {
-    maxLabel <- "Min. Instantaneous"
-    minLabel <- "Max. Instantaneous"
+  #Validation Options
+  requiredTimeSeriesFields <- c(
+    "points",
+    "approvals",
+    "qualifiers",
+    "startTime",
+    "endTime",
+    "isVolumetricFlow",
+    "units",
+    "grades",
+    "type",
+    "gaps",
+    "gapTolerances",
+    "name"
+  )
+  requiredMetadataFields <- c(
+    'startDate',
+    'endDate',
+    'isInverted',
+    'timezone',
+    'stationId',
+    'stationName',
+    'title'
+  )
+
+  #Validate Report Metadata
+  metaData <- fetchReportMetadata(reportObject)
+
+  #Get Necessary Report Metadata
+  if(validateFetchedData(metaData, "metadata", requiredMetadataFields)){
+    timezone <- fetchReportMetadataField(reportObject, 'timezone')
+    excludeMinMaxFlag <- parseReportMetadataField(reportObject, 'excludeMinMax', FALSE)
+    invertedFlag <- parseReportMetadataField(reportObject, 'isInverted', FALSE)
+    startDate <- toStartOfMonth(flexibleTimeParse(fetchReportMetadataField(reportObject, 'startDate'), timezone=timezone))
+    endDate <- toEndOfMonth(flexibleTimeParse(fetchReportMetadataField(reportObject, 'endDate'), timezone=timezone))
+    date_seq_mo <- seq(from=startDate, to=endDate, by="month")
+    first_yr <- date_seq_mo[which(month(date_seq_mo) == 1)[1]]
+    date_seq_yr <- seq(from=first_yr, to=endDate, by="year")
+    month_label_location <- date_seq_mo + (60*60*24*14) #make at 15th of month
+    month_label_split <- strsplit(as.character(month(date_seq_mo, label=TRUE)), "")
+    month_label <- unlist(lapply(month_label_split, function(x) {x[1]}))
   }
-  
-  if(anyDataExist(fiveyrData)){
-    
-    fiveyrInfo <- parseFiveYrSupplemental(data, fiveyrData)
-    
-    plot_object <- gsplot(yaxs = 'i', xaxt = "n", mar = c(8, 4, 4, 2) + 0.1) %>%
-      axis(side = 1, at = fiveyrInfo$date_seq_mo, labels = FALSE) %>%
-      view(xlim = c(fiveyrInfo$startDate, fiveyrInfo$endDate)) %>%
+
+  #Get Priority Stat Time Series
+  priorityStat <- getPriorityStat(reportObject)
+
+  #Get Basic Plot data
+  statTimeSeries <- parseTimeSeries(reportObject, priorityStat[['data_nm']], priorityStat[['descr_nm']], timezone, isDV=TRUE)
+  statTimeSeriesEst <- parseTimeSeries(reportObject, priorityStat[['data_nm']], priorityStat[['descr_nm']], timezone, estimated=TRUE, isDV=TRUE)
+
+  #Validate Basic Plot Data
+  if(is.null(c(statTimeSeries, statTimeSeriesEst))){
+    return(NULL)
+  }
```

Comment:
I'm normally opposed to multiple return calls, but in this case :+1: because we'd want to skip all the other plotting stuff anyways. And early returns can be much cleaner than large if-else statements